src/app/modules/search-input/components/advanced-search/advanced-search.component.ts

```diff
@@ -35,6 +39,7 @@ export class AdvancedSearchComponent<T> implements OnInit {
 
   @Output() paramsChange = new EventEmitter<QueryFilters<T>>();
   @Output() switchToBasic = new EventEmitter<void>();
+  @Output() runSearch = new EventEmitter<SearchQuery<T> | void>();
```

Comment:
There seem to be two conflicts with current architecture:
1. By emitting a query in `runSearch` some of the functionality of `paramsChange` is duplicated.
2. Advanced search was designed not to know that it is advanced, i.e. to emit QueryFilters instead of `SearchQuery`. The idea here was that maybe in the future we could use advanced and basic search components separately.

I think it would be a good idea to change this to always emit void and to rely on `paramsChange` for new params.